File: xap-drools-integration-master/common/src/main/java/com/c123/demo/real/BaseAccountOperationFact.java
package com.c123.demo.real;

import java.io.Serializable;
import java.math.BigDecimal;
import java.util.Date;

public class BaseAccountOperationFact extends BaseCustomerFact implements Serializable {


	   private BigDecimal actualAmount;
	    
		public BaseAccountOperationFact() {
			super();
		}    
	    
	
		public BigDecimal getActualAmount() {
			return actualAmount;
		}
		public void setActualAmount(BigDecimal actualAmount) {
			this.actualAmount = actualAmount;
		}


		@Override
		public String toString() {
			return "BaseAccountOperationFact [actualAmount=" + actualAmount
					+ "]";
		}


		
}

File: Drools_client/solution/Drools.Common/Entities/BaseAccountOperationFact.cs
using GigaSpaces.Core.Metadata;
using System;

namespace Drools.Common.Entities
{
    [Serializable]
    [SpaceClass(AliasName = "com.c123.demo.real.BaseAccountOperationFact")]
    public class BaseAccountOperationFact : BaseCustomerFact
    {
        [SpaceProperty(AliasName = "actualAmount")]
        public decimal? ActualAmount { get; set; }


        public BaseAccountOperationFact()
        {

        }

        public BaseAccountOperationFact(BaseAccountOperationFact originFact)
            : base(originFact)
        {
            ActualAmount = originFact.ActualAmount;
            DateTime = originFact.DateTime;
        }


        // To verify that we will not duplicate facts, this is the unique definition of a fact.
        protected override object[] IdComponents
        {
            get
            {
                return new object[]
                {
                    this.NetworkID, this.RequestReference
                };
            }
        }
    }
  
}
